simulator/examples/triangle-contains.rs

```diff
@@ -0,0 +1,130 @@
+//! # Example: Triangles and `contains()`
+//!
+//! Shows multiple triangles with different properties in green and a crosshair under the mouse
+//! cursor in cyan. If the cursor enters a triangle, that triangle will turn red.
+
+use embedded_graphics::{
+    pixelcolor::Rgb888,
+    prelude::*,
+    primitives::{Line, Triangle},
+    style::PrimitiveStyle,
+};
+use embedded_graphics_simulator::{
+    OutputSettingsBuilder, SimulatorDisplay, SimulatorEvent, Window,
+};
+use sdl2::keyboard::Keycode;
+
+const PAD: i32 = 10;
+const SPACING: i32 = 66;
+
+fn empty_crosshair(point: Point, color: Rgb888, display: &mut SimulatorDisplay<Rgb888>) {
+    let radius = Size::new_equal(4);
+    let inner_radius = Size::new_equal(2);
+
+    Line::new(point - radius.x_axis(), point - inner_radius.x_axis())
+        .points()
+        .chain(Line::new(point + radius.x_axis(), point + inner_radius.x_axis()).points())
+        .chain(Line::new(point - radius.y_axis(), point - inner_radius.y_axis()).points())
+        .chain(Line::new(point + radius.y_axis(), point + inner_radius.y_axis()).points())
+        .map(|p| Pixel(p, color))
+        .draw(display)
+        .unwrap();
```

Comment:
Should we use iterator chaining in the examples? It's a nice feature, but breaks HW acceleration.